### PR TITLE
[opt](function) opt ParseUrl function by process the input const column

### PR DESCRIPTION
**Pull request by Mryange**
## Proposed changes

Issue Number: close #xxx

before
```
mysql [test10]>select count(parse_url(url, 'HOST')) from urldb;
+-------------------------------+
| count(parse_url(url, 'HOST')) |
+-------------------------------+
|                      10000000 |
+-------------------------------+
1 row in set (1.12 sec)
```

now
```
mysql [test10]>select count(parse_url(url, 'HOST')) from urldb;
+-------------------------------+
| count(parse_url(url, 'HOST')) |
+-------------------------------+
|                      10000000 |
+-------------------------------+
1 row in set (0.82 sec)
```

### CODE REVIEW

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

File: be/src/vec/functions/function_string.h [hunk 1]
```diff
@@ -2277,70 +2279,102 @@ class FunctionStringParseUrl : public IFunction {
         res_offsets.resize(input_rows_count);
 
         size_t argument_size = arguments.size();
-        bool has_key = argument_size >= 3;
+        const bool has_key = argument_size >= 3;
```
Comment (HappenLee) [inline be/src/vec/functions/function_string.h:2282]: max size seems 3，should be dcheck bigger than 3？

File: be/src/vec/functions/function_string.h [hunk 2]
```diff
@@ -2277,70 +2279,102 @@ class FunctionStringParseUrl : public IFunction {
...
+    }
+    template <bool url_const, bool part_const, bool key_const>
+    static Status vector_parse_key(const ColumnString* url_col,
+                                   std::vector<UrlParser::UrlPart>& url_parts,
+                                   const ColumnString* key_col, const int size,
+                                   ColumnUInt8::Container& null_map_data,
+                                   ColumnString::Chars& res_chars,
+                                   ColumnString::Offsets& res_offsets) {
+        for (size_t i = 0; i < size; ++i) {
+            UrlParser::UrlPart& url_part = url_parts[part_const ? 0 : i];
```
Comment (HappenLee) [inline be/src/vec/functions/function_string.h:2366]: use index_check_const

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by Gabriel39 — APPROVED** [review]

**Pull request merged**